src/ngraph_builder.cc

```diff
@@ -3332,32 +3332,46 @@ static Status TranslateSplitVOp(
   std::vector<int> lengths;
   TF_RETURN_IF_ERROR(GetStaticInputVector(op, 1, static_input_map, &lengths));
 
-  int length = 0;
-  int idx = -1;
-  for (int i = 0; i < lengths.size(); ++i) {
-    if (lengths[i] != -1) {
-      length += lengths[i];
-    } else {
-      idx = i;
-    }
-  }
-
   ng::Shape shape = ng_input->get_shape();
   int rank = shape.size();
   std::vector<size_t> lower(rank, 0);
   std::vector<size_t> upper(shape);
-
   std::vector<int> split_dim_vec;
   TF_RETURN_IF_ERROR(
       GetStaticInputVector(op, 2, static_input_map, &split_dim_vec));
 
   int split_dim = split_dim_vec[0] + (split_dim_vec[0] < 0 ? (int64)rank : 0);
 
-  // Size splits must sum to the dimension of value along split_dim
-  if (idx > 0) {
+  bool has_one_neg = false;
+  
+  // length: Length of size_splits
+  int length = 0;
+  int idx = -1;
+
+  // Find out the total length of the splits and locate it's index
+  for (int i = 0; i < lengths.size(); ++i) {
+    if (lengths[i] != -1) {
+      length += lengths[i];
+    } else {
+      if(has_one_neg) {
+        return errors::InvalidArgument(
+          "size_splits can only have one -1");
+      } else {
+        idx = i;
+        has_one_neg = true;
+      }
+    }
+  }
+
+  if(idx > 0) {
     lengths[idx] = shape[split_dim] - length;
   }
 
+  if((idx == -1 && length != shape[split_dim]) || (idx > 0 && lengths[idx] <= 0)) {
+          return errors::InvalidArgument(
+          "The length of size_splits must sum to the dimension of value along split_dim");
```

Comment:
`split_dim` is an internal local variable. We should use `axis`, which is the TF name for this variable.
`dimension of value` -> did you mean `value of dimension`?

Also you could print out the value of `axis`

Ideally, the 2 error cases should get different error messages. Maybe we can split them into 2 `if`s, instead of `or`ing the error condition `((idx == -1 && length != shape[split_dim]) || (idx > 0 && lengths[idx] <= 0))`